swing-lib/src/test/java/org/triplea/swing/SwingActionTest.java

```diff
@@ -44,89 +47,80 @@
   private Consumer<KeyEvent> consumer;
 
   @Test
-  public void testActionOf() {
+  void testActionOf() {
     final Action action = SwingAction.of("Name1234", listener);
     assertEquals("Name1234", action.getValue(Action.NAME));
     action.actionPerformed(event);
     verify(listener).actionPerformed(event);
   }
 
   @Test
-  public void testKeyReleaseListener() {
+  void testKeyReleaseListener() {
     final KeyEvent event = mock(KeyEvent.class);
     final KeyListener action = SwingAction.keyReleaseListener(consumer);
     action.keyReleased(event);
     verify(consumer).accept(event);
   }
 
   @Test
-  public void testInvokeAndWait_ShouldInvokeActionWhenCalledOffEdt()
-      throws Exception {
+  void testInvokeAndWait_ShouldInvokeActionWhenCalledOffEdt() throws Exception {
     SwingAction.invokeAndWait(action);
 
     verify(action).run();
   }
 
   @Test
-  public void testInvokeAndWait_ShouldInvokeActionWhenCalledOnEdt()
-      throws Exception {
-    SwingUtilities.invokeAndWait(() -> {
-      try {
-        SwingAction.invokeAndWait(action);
-      } catch (final InterruptedException e) {
-        throw new RuntimeException(e);
-      }
-    });
+  void testInvokeAndWait_ShouldInvokeActionWhenCalledOnEdt() throws Exception {
+    SwingUtilities.invokeAndWait(() -> assertDoesNotThrow(() -> SwingAction.invokeAndWait(action)));
```

Comment:
I'm not sure this is a simplification, under what circumstances could the method ever throw? The point of this test is that we invoke something on EDT, not meant to test interruptibles throwing or not which should be close to an impossible condition for the test.

The only reason for the throwing of runtime exception at all was to handle the interrupted exception and that is just an artifact of using the lambda notation.

Reply:
The big change I see here is we wrap a throwing API with a non-throwing API. The assertion I think here is a bit misleading because it is API syntax sugar and not actually testing. The assertion specifically communicates that we are testing the throwing behavior, when actually the test does not care at all about the throwing behavior and an interrupted exception is practically an impossible condition (and is not at all what we are attempting to test).

Reply:
Thinking about this a moment longer, i think the problem I'd point to is that the name of the test method and what is being tested no longer agree.

Reply:
This test class I do think needs some help, it is hard to understand. If we invoke an action on or off EDT, what are we testing exactly? I'm pretty tempted to suggest that some of these test should just be pruned.